framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -116,13 +124,18 @@ public DependentTypesHelper(AnnotatedTypeFactory factory) {
         this.annoToElements = new HashMap<>();
         for (Class<? extends Annotation> expressionAnno : factory.getSupportedTypeQualifiers()) {
             List<String> elementList = getExpressionElementNames(expressionAnno);
-            if (elementList != null && !elementList.isEmpty()) {
+            if (!elementList.isEmpty()) {
                 annoToElements.put(expressionAnno, elementList);
             }
         }
-        this.expressionErrorChecker = new ExpressionErrorChecker();
-        this.standardizeTypeAnnotator = new StandardizeTypeAnnotator();
-        this.viewpointAdaptedCopier = new ViewpointAdaptedCopier();
+        this.hasDependentTypeScanner =
+                new SimpleAnnotatedTypeScanner<>(
+                        (type, unused) ->
+                                type.getAnnotations().stream().anyMatch(this::isExpressionAnno),
```

Comment:
Streams are likely to be less efficient than a for loop.  That is a small thing, but this is likely to get called a lot.